lib/mongoid/fields/standard.rb

```diff
@@ -231,7 +231,11 @@ def included?(fields)
       # @since 2.4.4
       def evaluated_default(doc)
         if default_val.respond_to?(:call)
-          evaluate_default_proc(doc)
+          begin
+            evaluate_default_proc(doc)
+          rescue NoMethodError => e
+            return nil
+          end
```

Comment:
@estolfo Do you mean we should move exceptional block to `initialize` method in `lib/mongoid/document.rb` ?